lib/rubocop/cop/performance/use_zip_to_wrap_array_contents.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Performance
+      # Checks for `.map { |id| [id] }` and suggests replacing it with `.zip`.
+      #
+      # @example
+      #   # bad
+      #   [1, 2, 3].map { |id| [id] }
+      #
+      #   # good
+      #   [1, 2, 3].zip
+      #
+      # @example
+      #   # good (no offense)
+      #   [1, 2, 3].map { |id| id }
+      #
+      # @example
+      #   # good (no offense)
+      #   [1, 2, 3].map { |id| [id, id] }
+      class UseZipToWrapArrayContents < Base
+        include RangeHelp
+        extend AutoCorrector
+
+        MSG = 'Use `.zip` instead of `.map { |id| [id] }`.'
+        RESTRICT_ON_SEND = %i[map].freeze
+
+        # Matches regular block form `.map { |e| [e] }`
+        # @!method map_with_array?(node)
+        def_node_matcher :map_with_array?, <<~PATTERN
+          (block
+            (send _ :map)
+            (args (arg _id))
+            (array (lvar _id)))
+        PATTERN
+
+        # Matches numblock form `.map { [_1] }`
+        # @!method map_with_array_numblock?(node)
+        def_node_matcher :map_with_array_numblock?, <<~PATTERN
+          (numblock
+            (send _ :map)
+            1
+            (array (lvar _))
+          )
+        PATTERN
+
+        def on_send(node)
+          parent = node.parent
```

Comment:
```suggestion
          return unless (parent = node.parent)
```

There is no parent when no receiver is given, this errors: `map { |id| [id] }`. Unlikely code to exist by itself (unless you maybe write a core extension) but with `ruby-lsp` it runs RuboCop on every keypress. Otherwise I have no complains anymore 👍

Reply:
Thanks, great catch! I added a spec for this. It also fails with a missing receiver, so I am adding the correct guard clause for that as well.